Frameworks/CoreGraphics/CGContext.mm

```diff
@@ -2253,55 +2191,77 @@ void CGContextClearRect(CGContextRef context, CGRect rect) {
                                  nullptr);
     }
 
-    ComPtr<ID2D1Image> currentImage{ image };
+    auto cleanup = wil::ScopeExit([this, layer]() {
+        if (layer) {
+            this->deviceContext->PopLayer();
+        }
 
-    ComPtr<ID2D1Effect> shadowEffect;
-    RETURN_IF_FAILED(_CreateShadowEffect(currentImage.Get(), &shadowEffect));
-    if (shadowEffect) {
-        RETURN_IF_FAILED(shadowEffect.As(&currentImage));
-    }
+        // TODO GH#1194: We will need to re-evaluate Direct2D's D2DERR_RECREATE when we move to HW acceleration.
+        this->PopEndDraw();
+    });
+
+    if (state.HasShadow()) {
```

Comment:
Additionally, it might be worthwhile to implement CGContextFlush at this juncture so that the bitmap context APIs can use it.